### ISSUE TITLE
Missing fields_for helper, and some forms weirdness

### ISSUE THREAD

**Issue by leafo** [ageweke/fortitude#21]
It should be possible to do

```ruby
form_for :test_form do |f|
  f.text_field "hello"
  f.fields_for :inner_thing do |inner|
    inner.text_field "greets"
  end
end
```

But `fields_for` is not a helper defined on `form_for` helper.

I added `fields_for` to the `output_yielded_methods`, which fixed it for my example above, generting the correct HTML:

```html
<form accept-charset="UTF-8" action="/test_form" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="j4Dax68aX/t7ajwBlpTTjeMT5MAaZjrNVYfE/yTW9ts=" /></div><input id="test_form_hello" name="test_form[hello]" size="30" type="text" /><input id="test_form_inner_thing_greets" name="test_form[inner_thing][greets]" size="30" type="text" /></form>
```

But, if I add an additional tag within the `fields_for` block, `inner.text_field "greets"` is no longer output.

```ruby
form_for :test_form do |f|
  f.text_field "hello"
  f.fields_for :inner_thing do |inner|
    inner.text_field "greets"
    div "hello"
  end
end
```

Outputs the HTML, (notice the div tag is there, but not the input)

```html
<form accept-charset="UTF-8" action="/test_form" method="post"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /><input name="authenticity_token" type="hidden" value="j4Dax68aX/t7ajwBlpTTjeMT5MAaZjrNVYfE/yTW9ts=" /></div><input id="test_form_hello" name="test_form[hello]" size="30" type="text" /><div>hello</div></form>
```

**Comment by ageweke** [comment]
What a seriously weird bug (i.e., adding another tag in the inner block makes the text field disappear).

I’ve written a spec that reproduces the issue, exactly as you said…working on fixing it now.

**Comment by ageweke** [comment]
OK, I understand exactly what's going on here, and how to fix it. I think I need to reify some of the helper definition logic in `lib/fortitude/widget/helpers.rb`. Proceeding…

**Comment by ageweke** [comment]
Just pushed the fix. Once it passes Travis, I'll release 0.9.4.

**Comment by ageweke** [comment]
Just released 0.9.4.
